### PR TITLE
Burst more bubble

### PR DESCRIPTION
**Pull request by eggrobin**
We will need to make this a `DoublePrecision<DegreesOfFreedom<Barycentric>>` for RealismOverhaul, but  this will require some library-writing. For now this will do.

### CODE REVIEW

File: ksp_plugin/part.cpp [hunk 1]
```diff
@@ -114,28 +114,24 @@ void Part::WriteToMessage(not_null<serialization::Part*> const message) const {
...
 
 not_null<std::unique_ptr<Part>> Part::ReadFromMessage(
     serialization::Part const& message) {
   // TODO(phl): Serialize/Deserialize the deletion callback.
-  not_null<std::unique_ptr<Part>> part =
-      make_not_null_unique<Part>(message.part_id(),
-                                 Mass::ReadFromMessage(message.mass()));
+  not_null<std::unique_ptr<Part>> part = make_not_null_unique<Part>(
+      message.part_id(),
+      Mass::ReadFromMessage(message.mass()),
```
Comment (pleroy) [inline ksp_plugin/part.cpp]: Please don't change the formatting.

File: ksp_plugin/part.cpp [hunk 2]
```diff
@@ -114,28 +114,24 @@ void Part::WriteToMessage(not_null<serialization::Part*> const message) const {
...
+      DegreesOfFreedom<Barycentric>::ReadFromMessage(message.degrees_of_freedom()),
+      /*TODO=*/nullptr);
```
Comment (pleroy) [inline ksp_plugin/part.cpp]: This statement is strangely formatted.

File: ksp_plugin/part.hpp [hunk 3]
```diff
@@ -47,7 +47,10 @@ class Part final {
  public:
   Part(PartId part_id,
        Mass const& mass,
+       DegreesOfFreedom<Barycentric> const& degrees_of_freedom,
        std::function<void()> deletion_callback);
+  // Calls the deletion callback passed at construction, if any.  Calls
```
Comment (pleroy) [inline ksp_plugin/part.hpp]: Blank line before.

File: ksp_plugin/part.hpp [hunk 4]
```diff
@@ -66,14 +69,10 @@ class Part final {
       Vector<Force, Barycentric> const& intrinsic_force);
   virtual Vector<Force, Barycentric> const& intrinsic_force() const;
 
-  // Clears, sets or returns the degrees of freedom of the part.  A part that is
-  // not in the bubble (i.e., that belongs to a pile-up that is not in the
-  // bubble) doesn't have degrees of freedom.
-  virtual void clear_degrees_of_freedom();
+  // Clears, sets or returns the degrees of freedom of the part.
```
Comment (pleroy) [inline ksp_plugin/part.hpp]: Remove "Clears".

File: ksp_plugin/pile_up.cpp [hunk 5]
```diff
@@ -23,40 +23,28 @@ using physics::DegreesOfFreedom;
...
 
-  RigidMotion<Bubble, Barycentric> const bubble_to_barycentric =
-      RigidMotion<Barycentric, Bubble>{
-          RigidTransformation<Barycentric, Bubble>{
-              bubble_barycentre.position(),
-              Bubble::origin,
-              OrthogonalMap<Barycentric, Bubble>::Identity()},
-          AngularVelocity<Barycentric>{},
-          bubble_barycentre.velocity()}.Inverse();
-  psychohistory_.Append(t, bubble_to_barycentric(calculator.Get()));
```
Comment (pleroy) [inline ksp_plugin/pile_up.cpp]: Why don't we append a point to the psychohistory now?

File: ksp_plugin/vessel.cpp [hunk 6]
```diff
@@ -92,6 +92,10 @@ void Vessel::FreeParts() {
   kept_parts_.clear();
 }
 
+not_null<Part*> Vessel::part(PartId id) const {
```
Comment (pleroy) [inline ksp_plugin/vessel.cpp]: const

Comment (eggrobin) [inline ksp_plugin/vessel.cpp]: Done.

File: ksp_plugin/pile_up.cpp (hunk 5 above)
Comment (eggrobin) [inline ksp_plugin/pile_up.cpp]: Oops.

File: ksp_plugin/part.hpp (hunk 4 above)
Comment (eggrobin) [inline ksp_plugin/part.hpp]: Done.

File: ksp_plugin/part.hpp (hunk 3 above)
Comment (eggrobin) [inline ksp_plugin/part.hpp]: Done.

File: ksp_plugin/part.cpp (hunk 2 above)
Comment (eggrobin) [inline ksp_plugin/part.cpp]: Done.

**Pull request merged**